src/main/java/ua/lviv/iot/algo/part1/lab6/service/DessertPlateService.java

```diff
@@ -0,0 +1,44 @@
+package ua.lviv.iot.algo.part1.lab6.service;
+
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+import ua.lviv.iot.algo.part1.lab6.models.DessertPlate;
+
+import java.util.*;
+
+@Service
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class DessertPlateService {
+    private Map<Integer, DessertPlate> dessertPlates = new HashMap<>();
+    private int nextAvailableId = 1;
+
+    public DessertPlate addDessertPlate(DessertPlate dessertPlate) {
+        if (dessertPlate.getId() == null) {
+            dessertPlate.setId((Integer) nextAvailableId++);
+        }
```

Comment:
this code should be executed all the time when you are trying to create new entity